### ISSUE TITLE
Interactive snapshot update/create mode

### ISSUE THREAD

**Issue by mrspartak** [vitest-dev/vitest#2229]
### Clear and concise description of the problem

We are testing media assets, by matching their md5 basically. So we have a suit, that basically calls the same render function but in different contexts (projects, parameters). Basically, the output must be validated before confirming, that this md5 should be a snapshot.
Jest has an interactive mode, but async describe is not a thing anymore, so we tried to switch to Vitest. 

### Suggested solution

1. Maybe do not create snapshots by default (I know, that env CI=1 solves it)
2. Have interactive update/create mode, that does 1 test at a time and asks to validate output

### Alternative

I could implement it myself but have no way to interrupt runner and it just times out. Also, libraries such as `enquirer` do not work during the run.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by sheremet-va** [comment]
> I could implement it myself but have no way to interrupt runner and it just times out. Also, libraries such as enquirer do not work during the run.

I think we can just run all the tests, and if there are snapshot mismatches, instead of an error we can generate new ones and show it in the terminal, like jest does, so user can confirm its content. After the change is confirmed, we mark it as resolved.

**Comment by mrspartak** [comment]
Yes, basically interrupt the runner until the action in the console is resolved (confirm or not the snapshot).

**Comment by mrspartak** [comment]
I just thought, would be cool, if `snapshotMatch` method would return if a snapshot is updated. We create media and would like to store confirmed in the cloud so we can also compare them later visually in case of a snapshot (which basically md5) mismatch.

**Comment by Ross-Esmond** [comment]
This is actually incredibly important to have.

Snapshot testing only catches bugs **if the developer is actually reading changes to the snapshots before triggering an update**, otherwise the developer is blindly updating the snapshots and a bug will pass right through. Without the interactive mode, it's difficult to vet each snapshot update in turn.

Say, for example, that I make a change which alters a couple dozen snapshots (it happens). Then say 95% of the snapshots are correct with only 5% indicating that something is wrong. At that point I have two choices: I can update all of the tests, in which case I lose track of which tests I found that were wrong and what they said, or I can update none of the tests, in which case I lose all of the progress I made checking and confirming the correctness of the 95%.

For anyone who is actually treating snapshot testing as "real" tests—as in, a test that will inform the developer if something is wrong—interactive mode is the only way to operate.

I'm not a contributor, so I don't have the right to say what should or shouldn't be worked on, but I would love it if this could get a P3 priority level, even if it's not marked for the version 4 milestone.

**Comment by mzhubail** [comment]
@Ross-Esmond,  I totally agree.

Once a project gets big enough, it can be a bit overwhelming to get through much snapshots manually (even if you update all, you'll still have to go through it manually).

I'll give a try at implementing it and might raise a PR soon.

**Comment by johtso** [comment]
Would love a feature like this!

**Comment by uncenter** [comment]
Some prior art from the Rust testing ecosystem is Insta's `review` mode: https://insta.rs/docs/cli/#review — I find myself frequently missing this feature from Insta!

**Comment by macarie** [comment]
Insta's `review` mode is exactly what I had in mind for this issue 😄
